### PR TITLE
Support empty-bodied computation expressions

### PR DESCRIPTION
**Pull request by brianrourkeboll**
## Description

- Add support for empty-bodied computation expressions.
  - Language suggestion: https://github.com/fsharp/fslang-suggestions/issues/1232
  - RFC FS-1144: https://github.com/fsharp/fslang-design/pull/774 (which see for more explanation of why this specific approach was taken)
  - RFC discussion: https://github.com/fsharp/fslang-design/discussions/775

### Examples

```fsharp
builder { }
```

```fsharp
seq { }
```

## Checklist

- [x] Test cases added.
- [x] Release notes entry updated.

> [!IMPORTANT]
> Shoutout to @edgarfgp @fabulous-dev for sponsoring my work on this.

### CODE REVIEW

**Comment by vzarytovskii** [comment]
Haven't reviewed it yet, but thinking out loud:

How would this play with resumable state machines. I assume it should be fine since it's just a transformation essentially.

And second, orthogonal comment - this feature will encourage more CE use, which my itself is not bad, but might lead to more nested CEs, which will involve more checking = being slower. We really need to do something with it. I made an attempt or two in the past, but change surface was bigger than I anticipated.

**Comment by brianrourkeboll** [comment]
> How would this play with resumable state machines. I assume it should be fine since it's just a transformation essentially.

That's a good call-out. I'll add some tests to make sure.

**Comment by psfinaki** [comment]
Cool stuff! Hope to give it a review soon. Can be a nice and intuitive addition to the language.

File: tests/FSharp.Compiler.ComponentTests/Conformance/Expressions/ComputationExpressions/ComputationExpressions.fs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+            |> compile
+            |> shouldFail
+            |> withErrorCode 789
+            |> withErrorMessage "'{ }' is not a valid expression. Records must include at least one field. Empty sequences are specified by using Seq.empty or an empty list '[]'."
+
+    /// F# 9.0 and above support empty-bodied computation expressions.
+    module Supported =
+        /// The language version that supports empty-bodied computation expressions.
+        /// TODO: Update this to the appropriate version when the feature comes out of preview.
+        let [<Literal>] SupportedLanguageVersion = "preview"
```
Comment (T-Gro) [inline tests/FSharp.Compiler.ComponentTests/Conformance/Expressions/ComputationExpressions/ComputationExpressions.fs:155]: I like this test-design here, will minimize PR diff once F#/NET 9  is released.

File: docs/release-notes/.Language/preview.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 * Allow returning bool instead of unit option for partial active patterns. ([Language suggestion #1041](https://github.com/fsharp/fslang-suggestions/issues/1041), [PR #16473](https://github.com/dotnet/fsharp/pull/16473))
 * Allow #nowarn to support the FS prefix on error codes to disable warnings ([Issue #17206](https://github.com/dotnet/fsharp/issues/16447), [PR #17209](https://github.com/dotnet/fsharp/pull/17209))
 * Allow ParsedHashDirectives to have argument types other than strings ([Issue #17240](https://github.com/dotnet/fsharp/issues/16447), [PR #17209](https://github.com/dotnet/fsharp/pull/17209))
+* Support empty-bodied computation expressions. ([Language suggestion #1232](https://github.com/fsharp/fslang-suggestions/issues/1232), [PR #17352](https://github.com/dotnet/fsharp/pull/17352))
```
Comment (T-Gro) [inline docs/release-notes/.Language/preview.md:11]: After it's released, I think linking to the RFC url will make more sense for readers (it's better structured and has more cases covered)

**Review by T-Gro — APPROVED** [review]

**Review by psfinaki — APPROVED** [review]
Nice work, great testing. I am not a CE guru (yet) so letting @vzarytovskii to make a final review here.

**Review by edgarfgp — APPROVED** [review]
Thanks @brianrourkeboll. Awesome work

File: docs/release-notes/.Language/preview.md (hunk 2 above)
Comment (brianrourkeboll) [reply docs/release-notes/.Language/preview.md:11]: I linked the RFC PR in acb6302.

**Pull request merged**